qa/tasks/ceph_manager.py

```diff
@@ -453,8 +456,10 @@ def kill_osd(self, osd=None, mark_down=False, mark_out=False):
                                '--journal-path', JPATH.format(id=imp_osd),
                            ])
                            + " --op apply-layout-settings --pool " + pool).format(id=osd)
-                    output = imp_remote.sh(cmd, wait=True, check_status=False)
-                    if 'Couldn\'t find pool' in output:
+                    proc = imp_remote.run(args=cmd,
+                                          wait=True, check_status=False,
+                                          stderr=StringIO)
+                    if 'Couldn\'t find pool' in proc.stderr.getvalue():
                         continue
                     if proc.exitstatus:
```

Comment:
Oh, this is not handled when used *remote.sh*, to handle it is better to use try-catch:
```
try:
    errors = BytesIO()
    imp_remote.sh(cmd, wait=True, stderr=errors)
    if 'Couldn\'t find pool' in six.ensure_str(errors.getvalue()):
except CommandFailedError as e:
    raise Exception("ceph-objectstore-tool apply-layout-settings "
                             "failed with {status}".format(status=e.exitstatus))
```
Alternatively the `six.StringIO` can be used instead of combination of `io.BytesIO` and `six.ensure_str`.